index.bs

```diff
@@ -11766,34 +12032,146 @@ must be {{%IteratorPrototype%}}.
+
+When a [=default asynchronous iterator object=] is first created, its state is not given.
+
+Note: [=Default asynchronous iterator objects=] do not have [=class strings=]; when
+<code class="idl">Object.prototype.toString()</code> is called on a
+[=default asynchronous iterator object=] of a given [=interface=], the [=class string=] of the
+[=asynchronous iterator prototype object=] of that [=interface=] is used.
+
+
+<h5 id="es-asynchronous-iterator-prototype-object">Asynchronous iterator prototype object</h5>
+
+The
+<dfn id="dfn-asynchronous-iterator-prototype-object" export>asynchronous iterator prototype object</dfn>
+for a given [=interface=] is an object that exists for every interface that has an
+[=asynchronously iterable declaration=].
+It serves as the prototype for [=default asynchronous iterator objects=] for the interface.
+
+The \[[Prototype]] [=internal slot=] of an [=asynchronous iterator prototype object=] must be
+{{%AsyncIteratorPrototype%}}.
+
+<div algorithm="to invoke the next property of asynchronous iterators">
+
+    An [=asynchronous iterator prototype object=] must have a <code class="idl">next</code> data
+    property with attributes { \[[Writable]]: <emu-val>true</emu-val>,
+    \[[Enumerable]]: <emu-val>true</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
+    and whose value is a [=built-in function object=] that behaves as follows:
+
+    1.  Let |interface| be the [=interface=] for which the
+        [=asynchronous iterator prototype object=] exists.
+
+    1.  Let |thisValidationPromiseCapability| be [=!=] [$NewPromiseCapability$]({{%Promise%}}).
+
+    1.  Let |object| be the result of calling [$ToObject$] on the
+        <emu-val>this</emu-val> value.
+
+    1.  [$IfAbruptRejectPromise$](|object|, |thisValidationPromiseCapability|).
+
+    1.  If |object| [=is a platform object=], then [=perform a security check=], passing:
+        *   the platform object |object|,
+        *   the identifier "<code>next</code>", and
+        *   the type "<code>method</code>".
+
+        If this threw an exception |e|, then:
+        1.  Perform [=!=] [$Call$](|thisValidationPromiseCapability|.\[[Reject]],
+            <emu-val>undefined</emu-val>, « |e| »).
+        1.  Return |thisValidationPromiseCapability|.\[[Promise]].
+
+    1.  If |object| is not a [=default asynchronous iterator object=] for |interface|, then:
+        1.  Issue: [=Realm=] check?
+        1.  Let |error| be a new {{ECMAScript/TypeError}}.
+        1.  Perform [=!=] [$Call$](|thisValidationPromiseCapability|.\[[Reject]],
+            <emu-val>undefined</emu-val>, « |error| »).
+        1.  Return |thisValidationPromiseCapability|.\[[Promise]].
+
+    1.  Let |nextSteps| be the following steps:
+        1.  Let |nextPromiseCapability| be [=!=] [$NewPromiseCapability$]({{%Promise%}}).
+        1.  Let |oldState| be |object|'s [=default asynchronous iterator object/state=].
+        1.  If |oldState| is null, then:
```

Comment:
So the state management here is a bit wonky.  Nothing before this point said anything about the state getting _set_ to null.  How would it get there?  And the initial state bit is not really set anywhere, and explicitly labeled as "not given" at the start.  I guess the intent is that "not given" is not the same thing as "null", but it's not very clear.  It would be good to be very clear about the value space of the state (IDL values, ES values, some other set of values) and when it's set to which of those.  It sounds like the intent is that it's in some other value space, which contains all possible opaque values, and two magical values "start of iteration" ("not given") and "stop iteration" ("null")?